### ISSUE TITLE
Wrong index used when migrating Ledger Nano from legacy IOTA to Firefly (small bug)

### ISSUE THREAD

**Issue by PatMulligan** [iotaledger/firefly#1381]
using **Firefly-desktop-1.2.0.AppImage for Linux**

I just successfully migrated my Ledger (Nano S) balance from the legacy IOTA network to the new Chrysalis network using the Firefly app. I encountered a small bug:

Starting at the step for **Choose Ledger account index**:

1. I input the index that had my balance which was not the default, 0.

2. It took a long time to load, although the Ledger Nano S screen was repeatedly showing "generating address" on and off.

3. Enough time passed that a notice popped up saying it was taking a long time and to make sure the device said "generating address", if not then unplug it and maybe reinstall the legacy app

4. Because it was showing "generating address" on and off, I thought maybe it was encountering an issue, so I unplugged and tried again. It took a while, then I thought maybe I'd just try the default index 0.

5. When I input the default index, 0, it showed me my correct balance. This is the bug :bug: because the balance was in fact on the previous index I was trying. It seems as though it retro-loaded the balance, even though I had input a different index, 0.

6. Thinking I was mistaken and it was on the 0 index, I proceeded through the steps, which brought me to the final migration validation. I.e., it showed me the separate transactions and it had me verify the attempt at the first one by matching the transaction codes.

7. After this I simply got an error that said something like "Something went wrong". I eventually tried from the beginning, using 0 as the index again, at which point it showed me there was no balance.

8. I tried my original index and waited a sufficient amount of time (same behavior on the device itself, showing generating address on and off) and it eventually showed me the correct balance. I proceeded and the migration went smoothly.

It's a small inconsequential bug, but it did scare me for a bit when I thought it was on the 0 index and it turned out not to be. Figured I would report it. Let me know if you need any more info.

All the best, and thanks for you guys' hard work!

**Comment by maxwellmattryan** [comment]
@PatMulligan Hmm this is interesting (also __thank you__ for the descriptive bug info)... did you ever get an error saying "Invalid address-index pair" or something like that (I think it's in step 6 where you get the "Something went wrong" error message perhaps)? It's possible that as long as you never refreshed or re-opened the Firefly app that it held onto previously received addresses somehow.

Each time the Ledger flashes on / off is when it is generating another address, which is totally expected. It can take a long time for the Ledger device to generate all the addresses to find your funds, so it's likely you just didn't wait long enough the first time for it to find all of your addresses. Will still investigate the bug though - nice find!

**Comment by PatMulligan** [comment]
@maxwellmattryan While I can't be *entirely* sure, I don't recall seeing anything along the lines of "Invalid address-index pair." If it were a dialog box that I would have had to respond to I can say pretty confidently that it did not appear though.

Yes, that was indeed the case. I imagine it must have been *just* before it would have successfully found the funds that I unplugged the IOTA  and re-tried it with the default (incorrect) index leading to the error. When I tried the correct index and gave it enough time everything went smoothly :)

Appreciate you guys' hard work!!
